modules/apps/27-interchain-accounts/host/keeper/keeper.go

```diff
@@ -151,19 +159,28 @@ func (k Keeper) getAppMetadata(ctx sdk.Context, portID, channelID string) (icaty
 }
 
 // GetActiveChannelID retrieves the active channelID from the store keyed by the provided connectionID and portID
-func (k Keeper) GetActiveChannelID(ctx sdk.Context, connectionID, portID string) (string, bool) {
-	store := ctx.KVStore(k.storeKey)
+func (k Keeper) GetActiveChannelID(ctx context.Context, connectionID, portID string) (string, bool) {
+	store := k.storeService.OpenKVStore(ctx)
 	key := icatypes.KeyActiveChannel(portID, connectionID)
 
-	if !store.Has(key) {
+	has, err := store.Has(key)
+	if err != nil {
+		panic(err)
+	}
+	if !has {
 		return "", false
 	}
 
-	return string(store.Get(key)), true
+	bz, err := store.Get(key)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(bz), true
```

Comment:
I think we can optimise this for gas consumption similarly to one or two funcs on the other PR. cc @DimitrisJim